ext/storage_sources/gcp_store/test/test_gcp_connection.cpp

```diff
@@ -27,8 +27,261 @@
  */
 
 #define CATCH_CONFIG_MAIN
+#include "gcp_connection.h"
+
 #include <catch2/catch.hpp>
+#include <typeinfo>
 
-#include "gcp_connection.h"
+namespace gcs = google::cloud::storage;
+using namespace gcs;
+
+static std::string
+create_file(const std::string file_name, const std::string payload)
+{
+    std::ofstream file(file_name);
+    file << payload;
+    file.close();
+    return file_name;
+}
+
+static auto
+upload_file(gcs::Client client, std::string bucket_name, const std::string bucket_prefix,
+  const std::string file_name, const std::string object_name)
+{
+    auto metadata = client.UploadFile("./" + file_name, bucket_name, bucket_prefix + object_name);
+
+    return metadata;
+}
+
+static bool
+check_exist(gcs::Client client, std::string bucket_name, const std::string bucket_prefix,
+  const std::string object_name)
+{
+    auto metadata = client.GetObjectMetadata(bucket_name, bucket_prefix + object_name);
+
+    // Metadata ok implies that the file is present.
+    return metadata.ok();
+}
+
+static bool
+check_object_num(
+  gcs::Client client, std::string bucket_name, const std::string bucket_prefix, const int num)
+{
+    auto objects_iterator = client.ListObjects(bucket_name, gcs::Prefix(bucket_prefix));
+    return std::distance(objects_iterator.begin(), objects_iterator.end()) == num;
+}
+
+// Concatenates a random suffix to the prefix being used for the test object keys. Example of
+// generated test prefix: "gcptest/unit/2022-31-01-16-34-10/623843294/".
+static std::string
+generate_test_prefix(std::string prefix)
+{
+    char time_str[100];
+    std::time_t t = std::time(nullptr);
+
+    REQUIRE(std::strftime(time_str, sizeof(time_str), "%F-%H-%M-%S", std::localtime(&t)) != 0);
+
+    prefix += time_str;
+
+    // Create a random device and use it to generate a random seed to initialize the generator.
+    std::random_device my_random_device;
+    unsigned seed = my_random_device();
+    std::default_random_engine my_random_engine(seed);
+
+    prefix += '/' + std::to_string(my_random_engine());
+    prefix += "/";
+
+    return prefix;
+}
+
+TEST_CASE("Testing class gcpConnection", "gcp-connection")
+{
+    std::string test_bucket_name = "unit_testing_gcp";
+    std::string prefix = "gcptest/unit/";
+
+    // Set up the test environment.
+    std::string test_bucket_prefix = generate_test_prefix(prefix);
```

Comment:
nit: If we're calling this function `generate_test_prefix` I'd recommend defining the whole prefix inside the function rather than passing in `prefix` and modifying it. It'll make it easier to keep track of what's happening. I believe strings are also pass-by-reference so we're probably unintentionally mutating prefix as a side effect here.

Reply:
now prefix is being generated inside `generate_test_prefix` without passing in any arguments in [8039651](https://github.com/wiredtiger/wiredtiger/pull/8749/commits/8039651071462d3bd21d784fdb335f39875e742f)